### PR TITLE
HDS-1395 Release page

### PR DESCRIPTION
**Pull request by laurakarhu**
## Description
New doc page about releases

## Related Issue
[HDS-1395](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1395?atlOrigin=eyJpIjoiNTRhZWQxZmMzZTIwNDJlODgwMjNmMGEzYTg3YjE3OTgiLCJwIjoiamlyYS1zbGFjay1pbnQif])

## Motivation and Context
Inform users about upcoming features and previous releases

## How Has This Been Tested?
Running docsite locally

- [x] open links in new tab

[HDS-1395]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-1395?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: site/src/docs/about/releases.mdx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+### Dropdown refactoring
+- design updates to make it clearer
+- removing outdated downshift-library
+- improving accessibility
+- bug fixes
+- refactor code
+
+## Latest release
+
+<ExternalLink href="https://github.com/City-of-Helsinki/helsinki-design-system/releases/latest">Latest release </ExternalLink>
```
Comment (mikkojamG) [inline site/src/docs/about/releases.mdx:32]: I would say this needs `openInNewTab` property (not entirely sure why it does not do that as a default) 🤔

File: site/src/docs/about/releases.mdx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - <ExternalLink href="https://github.com/City-of-Helsinki/helsinki-design-system/releases/latest/download/hds-design-kit.zip">Design kit</ExternalLink>
+    - <ExternalLink href="https://github.com/City-of-Helsinki/helsinki-design-system/releases/latest/download/hds-favicon-kit.zip">Favicon kit </ExternalLink>
+    - <ExternalLink href="https://github.com/City-of-Helsinki/helsinki-design-system/releases/latest/download/hds-icon-kit.zip">Icon kit </ExternalLink>
+
+## All previous releases  
+
+See previous releases and their release notes in <ExternalLink href="https://github.com/City-of-Helsinki/helsinki-design-system/releases" >GitHub </ExternalLink>
```
Comment (mikkojamG) [inline site/src/docs/about/releases.mdx:39]: I would say this needs `openInNewTab` property 🤔

File: site/src/docs/about/releases.mdx (hunk 1 above)
Comment (laurakarhu) [reply site/src/docs/about/releases.mdx]: Added

File: site/src/docs/about/releases.mdx (hunk 2 above)
Comment (laurakarhu) [reply site/src/docs/about/releases.mdx]: Added

**Review by mikkojamG — APPROVED** [review]

**Review by harriplappalainen — APPROVED** [review]
All good! Great work 👍

**Pull request merged**